packages/plugin-amplitude-node/lib/index.ts

```diff
@@ -36,19 +41,27 @@ export class AmplitudePlugin extends Plugin {
     this.amplitude = new Amplitude(this.apiKey, this.options);
   }
 
-  identify(userId: string, properties?: Properties) {
-    this.amplitude!.identify({
+  async identify(userId: string, properties?: Properties, options?: EventOptions) {
+    const { callback } = (options?.metadata?.[this.id] ?? {}) as Partial<AmplitudeMetadata>;
```

Comment:
`options?.metadata?.[this.id] ?? {}`  is used a lot. I added a new protected method Plugin.getPluginMetadata() that we can use instead. I updated the Segment plugins to use the new method. Can you please update the others to match?

Reply:
Done